packages/client/src/base/model-initialization-constraint.ts

```diff
@@ -23,70 +23,72 @@ import { Action, Deferred, InitializeCanvasBoundsAction, SetModelAction, UpdateM
  * the GLSP model initialization is completed, that is, the model has been set,
  * its bounds have been computed, the canvas bounds are available, etc.
  *
- * An injectable implementation of this constraint will be used by the
- * `GLSPActionDispatcher` to determine when the initialization is completed.
- * The action dispatcher therefore provides a promise via `onceInitialized()`
- * to trigger clients that want to dispatch an action, once the initialization
- * is done.
- *
- * For most of the cases `DefaultInitializationConstraint` can be used. In fact,
- * it is bound by default. However, custom implementations can rebind other
- * implementations of this constraint to, for instance, delay further before the
- * `onceInitialized()` promise is fulfilled by the `GLSPActionDispatcher`.
+ * In the default implementation the initialization process will be completed once the
+ * `DiagramLoader` has received and processed the initial {@link SetModelAction}
+ * from the server.
  */
 @injectable()
-export abstract class ModelInitializationConstraint {
-    protected completion: Deferred<void> = new Deferred();
-    protected completed = false;
-
-    get isCompleted(): boolean {
-        return this.completed;
+export class ModelInitializationConstraint implements Disposable {
```

Comment:
I wonder if `ModelInitializationConstraint` is now still a good name. Shouldn't it now be `ModelIinitializationStatus` or something like that?